### PR TITLE
[SPARK-30302][SQL] Complete info for show create table for views

### PR DESCRIPTION
**Pull request by wzhfy**
### What changes were proposed in this pull request?

Add table/column comments and table properties to the result of show create table of views.

### Does this PR introduce any user-facing change?

When show create table for views, after this patch, the result can contain table/column comments and table properties if they exist.

### How was this patch tested?

add new tests

### CODE REVIEW

**Comment by wzhfy** [comment]
retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala [hunk 1]
```diff
@@ -1013,10 +1013,10 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
     builder ++= s"CREATE$tableTypeString ${table.quotedString}"
 
     if (metadata.tableType == VIEW) {
-      if (metadata.schema.nonEmpty) {
-        builder ++= metadata.schema.map(_.name).mkString("(", ", ", ")")
-      }
-      builder ++= metadata.viewText.mkString(" AS\n", "", "\n")
+      showViewDataColumns(metadata, builder)
+      showTableComment(metadata, builder)
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: how about renaming `showTableComment` to `showTableOrViewComment `?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala [hunk 2]
```diff
@@ -1029,6 +1029,35 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
     builder.toString()
   }
 
+  private def showViewDataColumns(metadata: CatalogTable, builder: StringBuilder): Unit = {
+    if (metadata.schema.nonEmpty) {
+      val viewColumns = metadata.schema.map { f =>
+        val comment = f.getComment()
+          .map(escapeSingleQuotedString)
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Can you add tests for quoted text cases?

File: sql/core/src/test/resources/sql-tests/inputs/show-create-table.sql [hunk 3]
```diff
@@ -64,3 +64,34 @@ DROP TABLE tbl;
...
+COMMENT 'This is a comment for view'
+AS SELECT a, b FROM tbl;
+
+SHOW CREATE TABLE view_SPARK_30302;
+DROP VIEW view_SPARK_30302;
+
+
+-- tblproperties
+CREATE VIEW view_SPARK_30302 (aaa, bbb)
+TBLPROPERTIES ('a' = '1')
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/inputs/show-create-table.sql]: Can you add tests for multiple prop cases?

File: sql/core/src/test/resources/sql-tests/results/show-create-table.sql.out [hunk 4]
```diff
@@ -245,3 +245,103 @@ DROP TABLE tbl
...
+-- !query 31 output
+
+
+
+-- !query 32
+SHOW CREATE TABLE view_SPARK_30302
+-- !query 32 schema
+struct<createtab_stmt:string>
+-- !query 32 output
+CREATE VIEW `view_SPARK_30302`(`aaa` COMMENT 'comment for aaa', `bbb`)
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/results/show-create-table.sql.out]: In hive, the format is like this;
```
hive> show create table t;
OK
CREATE TABLE `t`(
  `a` int COMMENT 'comment for a', 
  `b` int)
COMMENT 'a\'aaaaa'
...
```
A single column info per line looks more readable to me. WDYT?

Comment (wzhfy) [reply sql/core/src/test/resources/sql-tests/results/show-create-table.sql.out]: Yes I also think that's better, but should we change it in a separate pr? In this pr, let's keep the same format as current hive table and data source table. How about that?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala (hunk 1 above)
Comment (wzhfy) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: how about make it more simple, `showComment`?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala (hunk 2 above)
Comment (wzhfy) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: test added

File: sql/core/src/test/resources/sql-tests/inputs/show-create-table.sql (hunk 3 above)
Comment (wzhfy) [reply sql/core/src/test/resources/sql-tests/inputs/show-create-table.sql]: test added, thanks

File: sql/core/src/test/resources/sql-tests/results/show-create-table.sql.out (hunk 4 above)
Comment (maropu) [reply sql/core/src/test/resources/sql-tests/results/show-create-table.sql.out]: Yea, that looks ok.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala (hunk 1 above)
Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Looks ok.

**Review by maropu — APPROVED** [review]
cc: @dongjoon-hyun @HyukjinKwon

**Comment by wzhfy** [comment]
@dongjoon-hyun @HyukjinKwon do you have time to take a look? Sorry to disturb you on Holiday :)

**Comment by wzhfy** [comment]
ping @dongjoon-hyun @HyukjinKwon. Could you spare some time to review this?

**Review by HyukjinKwon — APPROVED** [review]
Looks fine.

**Comment by maropu** [comment]
Thanks for the check, @HyukjinKwon! Merged to master.

**Comment by maropu** [comment]
@wzhfy  do you have time to make a pr for https://github.com/apache/spark/pull/26944#discussion_r360193294?

**Comment by dongjoon-hyun** [comment]
Sorry for missing pings here, @maropu and @wzhfy. I was sway since Dec. 20th~

**Comment by maropu** [comment]
nvm, happy new year, @dongjoon-hyun ~

**Comment by wzhfy** [comment]
> @wzhfy do you have time to make a pr for [#26944 (comment)](https://github.com/apache/spark/pull/26944#discussion_r360193294)?

@maropu Yes, I will submit a pr in several days.

**Comment by maropu** [comment]
Thanks~

**Comment by dongjoon-hyun** [comment]
Happy new year, @maropu . 😄

**Pull request closed without merging**